[Lim Yong En Dominic] iP

Arixeyeion opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Arixeyeion committed:
Changed name from duke to zoe and changed function from displaying duke to required IP level 0 text.

Arixeyeion committed:
Added the echo capability, bot will only terminate when the command "bye" is given.

Arixeyeion committed:
Add and List directly implemented into main.

Arixeyeion committed:
added mark/unmark capability

Arixeyeion committed:
reverted "Zoe" to main, created Zoe outside of main and all dependencies based on level-4

Arixeyeion committed:
added some error handling for zoe. Still missing specific error handling for tasks concerning time like deadline and event.

Arixeyeion committed:
added the delete capability

Arixeyeion committed:
Add automated testing from week2

Arixeyeion committed:
Add saving capability through loader and saver classes. Upon launch, Zoe will read saved tasks from previous sessions and she will save new tasks from the current session.

Arixeyeion committed:
Fix loading issue

Arixeyeion committed:
Add simple date capability

Arixeyeion committed:
Add more oop. Change main to zoe.

Arixeyeion committed:
Remove main

Arixeyeion committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	data/SavedTasks.txt
#	src/main/java/Deadline.java

Arixeyeion committed:
Fix Ui printing out wrong information regarding index size

Arixeyeion committed:
Shift interacting classes to package except Zoe as in the main file.

Arixeyeion committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Arixeyeion committed:
Shift Saver to package, make constructor public.

Arixeyeion committed:
Add test files to test event and todo saves

Arixeyeion committed:
Delete duke from folder to avoid confusion. Adjust build.gradle to generate zoe jar files.

Arixeyeion committed:
Attempt to add javadoc

Arixeyeion committed:
Add javadocs to public classes and methods

Arixeyeion committed:
Add find capability

Arixeyeion committed:
Adjust code to abide by coding standard

Arixeyeion committed:
Merge branch 'branch-A-Javadoc', resolve conflicts

# Conflicts:
#	src/main/java/zoe/Parser.java
#	src/main/java/zoe/Ui.java

Arixeyeion committed:
Merge branch 'branch-level-9'

Arixeyeion committed:
Attempt checkstyle

Arixeyeion committed:
Change to GUI

KhoonSun47 commented on src/main/java/zoe/Loader.java:
```diff
@@ -0,0 +1,52 @@
+package zoe;
+
+import java.io.*;
```
Should you list your imported classes explicitly to follow the Java Coding Standard?

KhoonSun47 commented on src/main/java/zoe/Zoe.java:
```diff
@@ -0,0 +1,27 @@
+package zoe;
+public class Zoe{
+    private Parser parser;
+    private TaskList tasks;
```
I like that you have distinguish clearly between single-valued and multi-valued variables, nice!

KhoonSun47 commented on src/main/java/zoe/Parser.java:
```diff
@@ -0,0 +1,110 @@
+     * If invalid command like empty todo is given, calls the respective responses
+     * @param command
+     */
+    public String carryOutShortCommand(String command) {
```
It is good that your method name explains the entity to the reader accurately and at a sufficient level of detail, nice!

KhoonSun47 commented on src/main/java/zoe/Saver.java:
```diff
@@ -0,0 +1,46 @@
+     * Saves data from the current zoe instance to a path when zoe closes
+     * @param path
+     */
+    public void saveTo(String path) {
```
It is good that your methods and classes follow the Java Code Quality of using nouns for classes/variables and verbs for methods/functions, nice!

KhoonSun47 commented on src/main/java/zoe/TaskList.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    /**
+     * Checks if a task exists in the list given an index when isValid(index) is called.
```
I noticed that there is no line spacing in between the description and parameter sections for your Javadocs comments, should you consider following the Java Coding Standard? 

One of the comment does not have a method associated with it, should you consider removing it?

KhoonSun47 reviewed (commented):
Great work overall! Just a few minor refinements required to ensure alignment with Java Coding Standards and enhance overall code quality. Keep up the outstanding effort!

JerryWang0000 commented on src/main/java/zoe/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+
+    protected String date;
+    protected String formattedDate;
+    public Deadline(String desc) {
```
Will it be better if there is a blank line and no space at line 11?

JerryWang0000 commented on src/main/java/zoe/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+        LocalDate inputDate = LocalDate.parse(date);
+        this.formattedDate = inputDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+        this.type = "D";
+        this.isDone = false;
```
I like this naming

JerryWang0000 commented on src/main/java/zoe/Event.java:
```diff
@@ -0,0 +1,38 @@
+ */
+public class Event extends Task {
+    protected String from;
+    protected String to;
```
Would it be better to state explicitly that the two strings represent timestamp?

JerryWang0000 commented on src/main/java/zoe/Event.java:
```diff
@@ -0,0 +1,38 @@
+        this.from = str[1].split(" ",2)[1];
+        this.to = str[2].split(" ", 2)[1];
+        this.type = "E";
+        this.isDone = isDoneNumber.equals("1");
```
"this" might not be necessary here as there is no ambiguous reference.

JerryWang0000 commented on src/main/java/zoe/Loader.java:
```diff
@@ -0,0 +1,52 @@
+            if (directory.exists() && file.exists()) {
+                BufferedReader reader = new BufferedReader(new FileReader(file));
+                ArrayList<Task> loadedTasks = new ArrayList<Task>();
+                String line;
```
The guideline would suggest not creating the object without initialization, but I'm not sure how to write better here tbh.

JerryWang0000 commented on src/main/java/zoe/Task.java:
```diff
@@ -0,0 +1,64 @@
+ */
+public class Task {
+    protected String description;
+    protected String type;
```
Is it better to have these attributes final? As I think they can be confirmed upon initialization.

JerryWang0000 commented on src/main/java/zoe/Task.java:
```diff
@@ -0,0 +1,64 @@
+    /**
+     * Records whether a task is done or not for data loading
+     */
+    public int isDoneNumerical() {
```
I thought the return type was boolean. Is it better to start the name with a verb?

JerryWang0000 commented on src/main/java/zoe/Task.java:
```diff
@@ -0,0 +1,64 @@
+    /**
+     * Marks a task as done to reflect in zoe
+     */
+    public void markAsDone() {
```
I like this naming as it is clearer than my version.

JerryWang0000 commented on src/main/java/zoe/Ui.java:
```diff
@@ -0,0 +1,76 @@
+    /**
+     * Prints opening statement
+     */
+    public String saysHi() {
```
Is it better to simply write sayHi()?

JerryWang0000 reviewed (commented):
The progress is ahead of me so sorry I cannot comment on the graphic part, but I'm glad that the rest of the codes are "readable" and with clarity.

Arixeyeion committed:
Change path to launcher to create new jar file

Arixeyeion committed:
Late commit for assertions.

Assertions were implemented as a branch but not committed to before working on the CodingQuality branch.

Let's commit this version with some CodingQuality implementation first as practice for pull requests.
This way, I still get to see the effects of some merging conflicts and parallel pull requests.

Arixeyeion committed:
Add rest of the CodeQuality changes.

There were some magic numbers in the code so this commits rectify that problem.

Doing away with magic numbers makes it easier to debug in future and provides clarity in code.

Arixeyeion committed:
Merge pull request #2 from Arixeyeion/branch-A-CodeQuality

Add rest of the CodeQuality changes.

Arixeyeion committed:
Merge branch 'master' into branch-A-Assertions

Arixeyeion committed:
Merge pull request #1 from Arixeyeion/branch-A-Assertions

Late commit for assertions.

AjayArvind2207 commented on src/main/java/zoe/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+     * @param i
+     */
+    public void delete(int i) {
+        tasks.remove(Integer.valueOf(i) - 1);
```
I feel there is an inconsistent usage of 'this'. Some methods use this referencing while others do not

AjayArvind2207 commented on src/main/java/zoe/Task.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
```
Maybe you can consider using this.isDone as opposed to isDone to be more consistent with styling 
```suggestion
        return (this.isDone ? "X" : " ");
```

AjayArvind2207 commented on src/main/java/zoe/Parser.java:
```diff
@@ -0,0 +1,113 @@
+                }
+            }
+
+        } else {
+            if (command.equals("todo")) {
+                res += ui.todoDescription();
+            } else {
+                res += ui.invalidCommand();
```
I really like your method names and they are very descriptive!

AjayArvind2207 reviewed (commented):
This is very well written code! Aside from some nits in the naming convention, the code LGTM. Keep it up and I hope to see this project develop further!

Arixeyeion committed:
Add sorting functionality for tasks. Tasks are now sorted in the following order: Todos, Events, Deadlines where Deadlines are sorted based on date.

Arixeyeion committed:
Change background colour to differentiate between chatbot text and user input for better clarity. Also changed maximum size of chatbot text to allow for dynamic sizing based on output text.

Arixeyeion committed:
Merge branch 'branch-C-Sort'

# Conflicts:
#	data/SavedTasks.txt

Arixeyeion committed:
Add and modify more javadocs to public methods and objects

Arixeyeion committed:
Test more potential exceptions that users may face. Standardize how data is processed for more flexible reading from file and neater presentation of data. Add more JUnit test to todo and event.

Arixeyeion committed:
Add pause when user inputs "bye" so the chat bot can say bye properly. Change saving using bye to saving whenever a user input is parsed to avoid dataloss should the user choose to close the application using the "X" button.

Arixeyeion committed:
Update warning message to include find for case when user keys in "find " with the empty space.

Arixeyeion committed:
Add picture to docs for website creation

Arixeyeion committed:
Change deadline to read dates and times instead of just dates

Change Ui picture so that it reflects new functionality as stated above

Arixeyeion committed:
Trial docs for webpage

Arixeyeion committed:
Check picture on website

Arixeyeion committed:
Check image on website

Arixeyeion committed:
Trial  website image upload again

Arixeyeion committed:
Fix deadline date time error display message to match updated version

Arixeyeion committed:
Adjust Zoe to catch more potential user errors so that it can accurately advise the user on how to use it

Arixeyeion committed:
Update javadocs to reflect code that is made in reference to online source

Update README to let user know that keyword in find <keyword> is case sensitive

Arixeyeion committed:
Update zoe download location to the released version of the ip
